### DISCUSSION TITLE
Document verification

### DISCUSSION THREAD

**Discussion by X-RoNar-X** [msiemens/tinydb#523] [Q&A]
Hi guys, 

Do you think it can be possible to add in the library a way to accept an insert or and update only if the content of the request is validated by a json validation library like Cerberus ? 

I modified the table class and especially the _update_table method like below but I want to now if it existe a better way of doing this ? 

```      # Perform the table update operation
        updater(table)

        # Convert the document IDs back to strings.
        # This is required as some storages (most notably the JSON file format)
        # don't support IDs other than strings.

        def validate_document(doc):

            if not self._verifier.validate(doc):
                #print(self._verifier.errors)
                raise TypeError("Bad document form")
            else:
                return True

        tables[self.name] = {
            str(doc_id): doc
            for doc_id, doc in table.items()
            if validate_document(doc)
        }`

**Comment by msiemens** [comment]
Hey @X-RoNar-X,

I feel that adding a concept of validation to TinyDB is beyond the scope of what TinyDB should do. But you could write a [custom table class](https://tinydb.readthedocs.io/en/latest/extend.html#subclassing-tinydb-and-table) that implements your own validation. Unfortunately it's not really composable with other table class modifications, but if you're writing your own application, it's a relatively simple solution.
